vyper/compile_lll.py

```diff
@@ -63,10 +63,10 @@ def compile_to_assembly(code, withargs=None, break_dest=None, height=0):
         return ['DUP' + str(height - withargs[code.value])]
     # Setting variables connected to with statements
     elif code.value == "set":
-        if height - withargs[code.args[0].value] > 16:
-            raise Exception("With statement too deep")
         if len(code.args) != 2 or code.args[0].value not in withargs:
             raise Exception("Set expects two arguments, the first being a stack variable")
+        if height - withargs[code.args[0].value] > 16:
+            raise Exception("With statement too deep")
```

Comment:
What did shifting this statement do?

Reply:
If you check the one if statement checks len(code.args), where the other assumes it's already there. That's why I swapped the two around ;)

Reply:
Nice gotcha